### DISCUSSION TITLE
Reading links in an endless loop

### DISCUSSION THREAD

**Discussion by seoeaa** [safevideo/autollm#136] [Q&A] [answered]
![2023-11-08_06-08-23](https://github.com/safevideo/autollm/assets/3609621/d973daab-8c49-4ce8-8699-e127ddaba690)

**Comment by talhasarit** [comment, marked as answer]
Hello @seoeaa,

It is expected to take long time to parse all of the URL's with child uri's that your provided website has. If you want to only parse the provided website but not child links that website has, please use our WebPageReader class instead:

```python
from autollm.utils.web_page_reader import WebPageReader

reader = WebPageReader()
documents = reader.load_data('http://example.com/some-page')
```
